src/settings/mod.rs

```diff
@@ -5,64 +5,67 @@
 use std::env;
 
 use config::{Config, ConfigError, Environment, File};
+use logging::MozlogLogger;
 
 use deserialize;
+use serialize;
 
 #[cfg(test)]
 mod test;
 
-#[derive(Debug, Default, Deserialize)]
+#[derive(Debug, Default, Deserialize, Serialize)]
 pub struct AuthDb {
     #[serde(deserialize_with = "deserialize::base_uri")]
     pub baseuri: String,
 }
 
-#[derive(Debug, Default, Deserialize)]
+#[derive(Debug, Default, Deserialize, Serialize)]
 pub struct Aws {
+    #[serde(serialize_with = "serialize::hidden_or_not_set")]
```

Comment:
Can we add some test coverage to assert that all of these properties that are `hidden_or_not_set` get serialized correctly in both states?